### PR TITLE
[UR] disable hwloc support for UMF

### PR DESCRIPTION
**Pull request by igchor**
Hwloc package is not avalable on certain distros (e.g. RHEL). To avoid forcing users to install hwloc from sources, disable hwloc in UMF.

Based on: https://github.com/oneapi-src/unified-memory-framework/pull/642

### CODE REVIEW

**Comment by pbalcer** [comment]
@intel/llvm-reviewers-runtime @intel/unified-runtime-reviewers this is ready, please review.

**Review by maarquitos14 — APPROVED** [review]
LGTM.

File: sycl/cmake/modules/FetchUnifiedRuntime.cmake [hunk 1]
```diff
@@ -113,22 +116,25 @@ if(SYCL_PI_UR_USE_FETCH_CONTENT)
...
   # Due to the use of dependentloadflag and no installer for UMF and hwloc we need
   # to link statically on windows
   if(WIN32)
     set(UMF_BUILD_SHARED_LIBRARY OFF CACHE INTERNAL "Build UMF shared library")
     set(UMF_LINK_HWLOC_STATICALLY ON CACHE INTERNAL "static HWLOC")
+  else()
+    set(UMF_DISABLE_HWLOC ${SYCL_UMF_DISABLE_HWLOC} CACHE INTERNAL "disable hwloc for UMF")
   endif()
 
+
```
Comment (dm-vodopyanov) [inline sycl/cmake/modules/FetchUnifiedRuntime.cmake:137]: nit: extra line?

File: sycl/cmake/modules/FetchUnifiedRuntime.cmake [hunk 2]
```diff
@@ -113,22 +116,25 @@ if(SYCL_PI_UR_USE_FETCH_CONTENT)
...
+  set(UNIFIED_RUNTIME_TAG a985a81dc9ba8adfcc8b54e35ad287e97766fb3e)
 
   set(UMF_BUILD_EXAMPLES OFF CACHE INTERNAL "EXAMPLES")
   # Due to the use of dependentloadflag and no installer for UMF and hwloc we need
   # to link statically on windows
   if(WIN32)
     set(UMF_BUILD_SHARED_LIBRARY OFF CACHE INTERNAL "Build UMF shared library")
     set(UMF_LINK_HWLOC_STATICALLY ON CACHE INTERNAL "static HWLOC")
+  else()
+    set(UMF_DISABLE_HWLOC ${SYCL_UMF_DISABLE_HWLOC} CACHE INTERNAL "disable hwloc for UMF")
```
Comment (dm-vodopyanov) [inline sycl/cmake/modules/FetchUnifiedRuntime.cmake:134]: nit:
```suggestion
    set(UMF_DISABLE_HWLOC ${SYCL_UMF_DISABLE_HWLOC} CACHE INTERNAL "Disable hwloc for UMF")
```

**Review by dm-vodopyanov — APPROVED** [review]

**Review by kbenzie — APPROVED** [review]
UR LGTM

**Comment by pbalcer** [comment]
@intel/llvm-gatekeepers please merge

**Pull request merged**